airbyte-integrations/connectors/source-instagram/unit_tests/test_source.py

```diff
@@ -17,77 +13,66 @@
     SyncMode,
 )
 
+from .conftest import GRAPH_URL, account_url, get_source, mock_fb_account_response
 
-logger = logging.getLogger("airbyte")
 
-GRAPH_URL = resolve_manifest(source=SourceInstagram(config={}, catalog=None, state=None)).record.data["manifest"]["definitions"][
-    "base_requester"
-]["url_base"]
-
-account_url = f"{GRAPH_URL}/me/accounts?fields=id%2Cinstagram_business_account"
+logger = logging.getLogger("airbyte")
 
 
 account_url_response = {
     "data": [{"id": "page_id", "name": "Airbyte", "instagram_business_account": {"id": "instagram_business_account_id"}}],
-    "paging": {"cursors": {"before": "before", "after": "after"}},
+    # I'm not sure why we added paging originally, but it's presence can lead to trying to fetch
+    # additional business accounts which is not intended because we don't have mocks supplied for query params `after=after`
+    # "paging": {"cursors": {"before": "before", "after": "after"}},
 }
 
 
-def test_check_connection_ok(api, requests_mock, some_config):
-    requests_mock.register_uri("GET", account_url, [{"json": account_url_response}])
-    ok, error_msg = SourceInstagram(config=some_config, catalog=None, state=None).check_connection(logger, config=some_config)
-    assert ok
-    assert not error_msg
-
-
-def test_check_connection_empty_config(api):
-    config = {}
-    ok, error_msg = SourceInstagram(config=config, catalog=None, state=None).check_connection(logger, config=config)
+def mock_api(requests_mock, some_config):
+    fb_account_response = mock_fb_account_response("unknown_account", some_config, requests_mock)
 
-    assert not ok
-    assert error_msg
-
-
-def test_check_connection_invalid_config_future_date(api, some_config_future_date):
-    ok, error_msg = SourceInstagram(config=some_config_future_date, catalog=None, state=None).check_connection(
-        logger, config=some_config_future_date
+    requests_mock.register_uri(
+        "GET",
+        f"{GRAPH_URL}/me/accounts?" f"access_token={some_config['access_token']}&summary=true",
+        [fb_account_response],
     )
 
-    assert not ok
-    assert error_msg
-
 
-def test_check_connection_no_date_config(api, requests_mock, some_config):
+def test_check_connection_ok(requests_mock):
+    some_config = {"start_date": "2021-01-23T00:00:00Z", "access_token": "unknown_token"}
+    mock_api(requests_mock, some_config)
     requests_mock.register_uri("GET", account_url, [{"json": account_url_response}])
-    some_config.pop("start_date")
-    ok, error_msg = SourceInstagram(config=some_config, catalog=None, state=None).check_connection(logger, config=some_config)
 
+    ok, error_msg = get_source(config=some_config, state=None).check_connection(logger, config=some_config)
     assert ok
     assert not error_msg
 
 
-def test_check_connection_exception(api, config):
```

Comment:
This was a very confusing test. For reasons I can't figure out, this test ends up accidentally using mocks from other tests which causes the HTTP error we want to get triggered to not get invoked. 

I gave this some time, but couldn't figure out how to fix this and to retain the same amount of coverage, I instead added a mock server test to `integration/test_api.py` that mocks a 400 to simulate receiving an error.